Fails to compile when dependent crate uses log4rs with "default_features = false"

Error message:

```
error[E0433]: failed to resolve. Could not find `pattern` in `encode`
   --> /Users/i_m/.cargo/registry/src/github.com-1ecc6299db9ec823/log4rs-syslog-3.0.2/src/syslog.rs:348:61
    |
348 |                 .unwrap_or_else(|| Box::new(log4rs::encode::pattern::PatternEncoder::default())),
    |                                                             ^^^^^^^ Could not find `pattern` in `encode`

error[E0277]: the trait bound `log4rs::encode::Encode: std::marker::Sized` is not satisfied
   --> /Users/i_m/.cargo/registry/src/github.com-1ecc6299db9ec823/log4rs-syslog-3.0.2/src/syslog.rs:348:36
    |
348 |                 .unwrap_or_else(|| Box::new(log4rs::encode::pattern::PatternEncoder::default())),
    |                                    ^^^^^^^^ `log4rs::encode::Encode` does not have a constant size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `log4rs::encode::Encode`
    = note: required by `<std::boxed::Box<T>>::new`

error: aborting due to 2 previous errors

Some errors occurred: E0277, E0433.
For more information about an error, try `rustc --explain E0277`.
error: Could not compile `log4rs-syslog`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,12 +24,11 @@ bitflags = { version = "1.0", default_features = false }
 lazy_static = { version = "1.0", default_features = false }
 libc = { version = "0.2", default_features = false }
 log = { version = "0.4", default_features = false }
-log4rs = { version = "0.8", default_features = false }
+log4rs = { version = "0.8", default_features = false, features = ["pattern_encoder"] }
 serde = { version = "1.0", default_features = false, optional = true }
 serde_derive = { version = "1.0", default_features = false, optional = true }
 
 [dev-dependencies]
-glob = "0.2"
-log4rs = "0.8"
-tempfile = "3.0"
-which = "2.0"
+glob = { version = "0.2", default_features = false }
+tempfile = { version = "3.0", default_features = false }
+which = { version = "2.0", default_features = false }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 name = "log4rs-syslog"
 description = "Syslog appender for log4rs based on libc's syslog() function (*nix only)"
-version = "3.0.2"
+version = "3.0.3"
 authors = ["Ivan Mironov <mironov.ivan@gmail.com>"]
 license = "MIT/Apache-2.0"
 keywords = ["logger", "log", "logging", "nix", "syslog"]

